### ISSUE TITLE
Copula Fit Replication with SAS using Uniform Marginals

### ISSUE THREAD

**Issue by njalex22** [DanielBok/copulae#42]
Hi Daniel,

I am comparing the copula fit between copulae and SAS and I am observing different results. In the [linked notebook](https://github.com/njalex22/copula_test/blob/main/copula_replication.ipynb), I create random data and conduct the probability integral transform using different distributions for the clayton copula. Regardless of which distribution I choose to conduct probability integral transform, the theta parameter for Clayton is the same. This was not my expectation but conducting the same copula fitting in SAS with the exact same uniformly distributed marginals provides a bit of context. 

When fitting a copula in SAS, you can specify whether your marginals are already uniformly distributed, or if they need to converted to uniformly distributed via the empirical CDF. First, I specified the marginals as already uniformly distributed, with results [here](https://github.com/njalex22/copula_test/blob/main/SAS_Copula_uniform.pdf). As expected (and different than the jupyter notebook), the theta parameter of the Clayton copula changed based on which distribution I used to conduct probability integral transform. The case where the input marginals were transformed to uniformly distributed via the empirical CDF yielded the exact same theta as copulae, providing me confidence in the fitting done for copulae.

Next, I fit all the uniformly distributed marginals to the Clayton copula in SAS while specifying that the marginals needed to be converted to uniformly distributed via the empirical CDF. In this iteration, the theta parameter estimated for the clayton copula was the same in all cases and identical to the estimate of the copulae. Those results are found [here](https://github.com/njalex22/copula_test/blob/main/SAS_Copula_Emp.pdf).

In conclusion, it appears copulae is converting the marginals to uniformly distributed via the empirical CDF regardless of the whether the data is already uniformly distributed or not. One of the biggest decisions when fitting a copula is choice of marginal distribution which should affect the fit of the copula as demonstrated by SAS. Is there a parameter I should change in the fit function such that copulae does not automatically convert marginals to uniformly distributed which should result in a different fit based on the distributions chosen to conduct the probability integral transform? Thanks!

**Comment by DanielBok** [comment]
Good point. I'll add an option `to_pobs` in 0.7.3 which will disable the automatic conversion of the input data in the `fit` method. This will be similar to the `log_lik` method's signature.

A little background, when I first wrote this for research work, most of the other researchers didn't bother casting the input data to its uniform marginals. Thus the default from the onset was to do this casting automatically (with the exception of the `EmpiricalCopula`). Most of the time, I guess it's fine since even if the data was already in its CDF form because if the dataset is representative enough, converting it to its pseudo observations will usually yield the same results. But I can see scenarios where it fails and where a manual treatment of the data is preferred over the automatic conversion.

**Comment by njalex22** [comment]
Sounds good, thanks!
